### PR TITLE
Update the changelog for 0.5.

### PR DESCRIPTION
**Pull request by jlewi**
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/2855)

### CODE REVIEW

File: scripts/update-changelog.sh [hunk 1]
```diff
@@ -22,20 +22,21 @@
 set -o errexit
 set -o nounset
 set -o pipefail
+set -x
 
 GITHUB_TOKEN=${GITHUB_TOKEN:-"NO"}
```
Comment (lluunn) [inline scripts/update-changelog.sh]: any github token will work?

Comment (jlewi) [reply scripts/update-changelog.sh]: It should; its should for API rate limits.

File: docs_dev/releasing.md [hunk 2]
```diff
@@ -400,3 +400,15 @@ Releasing a new version on the website requires the following steps:
...
+
+1. After the release branch is created run the following script to update the changelog
+
+   ```
+   update-changelog.sh
+   ```
+
+1. Create a PR with the resulting changes.
+
+1. Repeate above steps as new release candidates are created
```
[the remark this answers is not in the document]
Comment (jlewi) [reply docs_dev/releasing.md]: Not yet.

I think we need some manual cleanup/refinement to make this more readable. I think the next step is to ask some folks from the PM group to review this.

[the remark this answers is not in the document]
Comment (lluunn) [reply docs_dev/releasing.md]: sounds good

**Pull request merged**